core/src/main/java/edu/wpi/grip/core/operations/network/ros/RealROSManager.java

```diff
@@ -0,0 +1,184 @@
+package edu.wpi.grip.core.operations.network.ros;
+
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import edu.wpi.grip.core.operations.network.NetworkPublisher;
+import org.ros.concurrent.CancellableLoop;
+import org.ros.internal.message.Message;
+import org.ros.namespace.GraphName;
+import org.ros.node.*;
+import org.ros.node.topic.Publisher;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentMap;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ *
+ */
+public class RealROSManager implements ROSManager {
```

Comment:
So then `ROSPublishOperation` should just take a Manager interface.
I did this so I could do class binding in the Guice module. I guess I can use named bindings instead.
In my mind, that screamed "not type safe" but if you'd rather me use named injection bindings I'll do that instead.